extensions/amazon-dynamodb/deployment/src/main/java/io/quarkus/dynamodb/deployment/DynamodbProcessor.java

```diff
@@ -1,58 +1,208 @@
+import java.net.URI;
 
+import org.jboss.jandex.DotName;
+import org.jboss.jandex.Type;
+
+import io.quarkus.arc.deployment.AdditionalBeanBuildItem;
+import io.quarkus.arc.deployment.BeanContainerBuildItem;
+import io.quarkus.arc.deployment.BeanRegistrationPhaseBuildItem;
+import io.quarkus.arc.processor.BuildExtension;
+import io.quarkus.arc.processor.InjectionPointInfo;
 import io.quarkus.deployment.annotations.BuildProducer;
 import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.annotations.ExecutionTime;
+import io.quarkus.deployment.annotations.Record;
 import io.quarkus.deployment.builditem.ExtensionSslNativeSupportBuildItem;
 import io.quarkus.deployment.builditem.FeatureBuildItem;
 import io.quarkus.deployment.builditem.JniBuildItem;
-import io.quarkus.deployment.builditem.substrate.ReflectiveClassBuildItem;
+import io.quarkus.deployment.builditem.ShutdownContextBuildItem;
 import io.quarkus.deployment.builditem.substrate.ServiceProviderBuildItem;
 import io.quarkus.deployment.builditem.substrate.SubstrateProxyDefinitionBuildItem;
+import io.quarkus.deployment.configuration.ConfigurationError;
+import io.quarkus.dynamodb.runtime.AwsApacheHttpClientConfig;
+import io.quarkus.dynamodb.runtime.AwsCredentialsProviderType;
+import io.quarkus.dynamodb.runtime.AwsNettyNioAsyncHttpClientConfig;
+import io.quarkus.dynamodb.runtime.DynamodbClientProducer;
+import io.quarkus.dynamodb.runtime.DynamodbConfig;
+import io.quarkus.dynamodb.runtime.DynamodbRecorder;
 import software.amazon.awssdk.http.SdkHttpService;
 import software.amazon.awssdk.http.apache.ApacheSdkHttpService;
 import software.amazon.awssdk.http.async.SdkAsyncHttpService;
 import software.amazon.awssdk.http.nio.netty.NettySdkAsyncHttpService;
+import software.amazon.awssdk.services.dynamodb.DynamoDbAsyncClient;
+import software.amazon.awssdk.services.dynamodb.DynamoDbClient;
+import software.amazon.awssdk.utils.StringUtils;
 
 public class DynamodbProcessor {
     public static final String AWS_SDK_APPLICATION_ARCHIVE_MARKERS = "software/amazon/awssdk";
 
-    @Inject
-    BuildProducer<ExtensionSslNativeSupportBuildItem> extensionSslNativeSupport;
+    private static final DotName SYNC_CLIENT_NAME = DotName.createSimple(DynamoDbClient.class.getName());
+    private static final DotName ASYNC_CLIENT_NAME = DotName.createSimple(DynamoDbAsyncClient.class.getName());
 
-    @BuildStep
-    void build(BuildProducer<FeatureBuildItem> feature) {
-        feature.produce(new FeatureBuildItem(FeatureBuildItem.DYNAMODB));
-    }
+    DynamodbConfig config;
 
     @BuildStep
     JniBuildItem jni() {
         return new JniBuildItem();
     }
 
     @BuildStep
-    void setupDynamoDb() {
+    SubstrateProxyDefinitionBuildItem httpProxies() {
+        return new SubstrateProxyDefinitionBuildItem("org.apache.http.conn.HttpClientConnectionManager",
+                "org.apache.http.pool.ConnPoolControl", "software.amazon.awssdk.http.apache.internal.conn.Wrapped");
```

Comment:
Are they useful in both cases? Looks related to the sync client only? If so maybe we should condition them?

Reply:
Fixed